File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/domain/model/Bank.java
package org.mifos.mobilewallet.mifospay.domain.model;

/**
 * Created by naman on 20/6/17.
 */

public class Bank {

    private final String name;
    private final int image;

    // 0= popular, 1 = other
    private final int type;

    public Bank(String name, int image, int type) {
        this.name = name;
        this.type = type;
        this.image = image;
    }

    public String getName() {
        return name;
    }

    public int getImage() {
        return image;
    }

    public int getType() {
        return type;
    }
}

File: core/model/src/commonMain/kotlin/org/mifospay/core/model/bank/Bank.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/mobile-wallet/blob/master/LICENSE.md
 */
package org.mifospay.core.model.bank

import org.mifospay.core.common.Parcelable
import org.mifospay.core.common.Parcelize

@Parcelize
data class Bank(
    val name: String,
    val image: Int,
    val bankType: BankType = BankType.OTHER,
) : Parcelable

enum class BankType {
    POPULAR,
    OTHER,
}
